tfhe/src/core_crypto/fft_128_split_impl/crypto/tests.rs

```diff
@@ -0,0 +1,215 @@
+    use super::super::math::fft::Fft128;
+    use crate::core_crypto::prelude::*;
+
+    type Scalar = u128;
+
+    // DISCLAIMER: these toy example parameters are not guaranteed to be secure or yield correct
+    // computations
+    // Define the parameters for a 4 bits message able to hold the doubled 2 bits message
+    let small_lwe_dimension = LweDimension(742);
+    let glwe_dimension = GlweDimension(1);
+    let polynomial_size = PolynomialSize(2048);
+    let lwe_modular_std_dev = StandardDev(sqr(0.000007069849454709433));
+    let glwe_modular_std_dev = StandardDev(sqr(0.00000000000000029403601535432533));
+    let pbs_base_log = DecompositionBaseLog(23);
+    let pbs_level = DecompositionLevelCount(1);
+
+    // Request the best seeder possible, starting with hardware entropy sources and falling back to
+    // /dev/random on Unix systems if enabled via cargo features
+    let mut boxed_seeder = new_seeder();
+    // Get a mutable reference to the seeder as a trait object from the Box returned by new_seeder
+    let seeder = boxed_seeder.as_mut();
+
+    // Create a generator which uses a CSPRNG to generate secret keys
+    let mut secret_generator =
+        SecretRandomGenerator::<ActivatedRandomGenerator>::new(seeder.seed());
+
+    // Create a generator which uses two CSPRNGs to generate public masks and secret encryption
+    // noise
+    let mut encryption_generator =
+        EncryptionRandomGenerator::<ActivatedRandomGenerator>::new(seeder.seed(), seeder);
+
+    println!("Generating keys...");
+
+    // Generate an LweSecretKey with binary coefficients
+    let small_lwe_sk =
+        LweSecretKey::generate_new_binary(small_lwe_dimension, &mut secret_generator);
+
+    // Generate a GlweSecretKey with binary coefficients
+    let glwe_sk =
+        GlweSecretKey::generate_new_binary(glwe_dimension, polynomial_size, &mut secret_generator);
+
+    // Create a copy of the GlweSecretKey re-interpreted as an LweSecretKey
+    let big_lwe_sk = glwe_sk.clone().into_lwe_secret_key();
+
+    // Generate the seeded bootstrapping key to show how to handle entity decompression,
+    // we use the parallel variant for performance reason
+    let std_bootstrapping_key = par_allocate_and_generate_new_seeded_lwe_bootstrap_key(
+        &small_lwe_sk,
+        &glwe_sk,
+        pbs_base_log,
+        pbs_level,
+        glwe_modular_std_dev,
+        seeder,
+    );
+
+    // We decompress the bootstrapping key
+    let std_bootstrapping_key: LweBootstrapKeyOwned<Scalar> =
+        std_bootstrapping_key.decompress_into_lwe_bootstrap_key();
+
+    // Create the empty bootstrapping key in the Fourier domain
+    let mut fourier_bsk = Fourier128LweBootstrapKey::new(
+        std_bootstrapping_key.input_lwe_dimension(),
+        std_bootstrapping_key.polynomial_size(),
+        std_bootstrapping_key.glwe_size(),
+        std_bootstrapping_key.decomposition_base_log(),
+        std_bootstrapping_key.decomposition_level_count(),
+    );
```

Comment:
no need for the seeded key, this was done mainly for example purposes

Reply:
what should i use instead?

Reply:
par_allocate without the seeded part and then you can skip the decompression